app/src/main/kotlin/com/mapzen/erasermap/view/MainActivity.kt

```diff
@@ -146,18 +151,32 @@ public class MainActivity : AppCompatActivity(), ViewController, Router.Callback
         LocationServices.FusedLocationApi?.requestLocationUpdates(locationRequest) {
             location: Location -> currentLocation = location
             val routeModeView = findViewById(R.id.route_mode) as RouteModeView
-            routeModeView.onLocationChanged(location)
+            if (routeModeView != null && routeModeView.getVisibility() == View.VISIBLE) {
```

Comment:
Good suggestion. Using view state to control the flow of data in the app is gross.

I added a commit that moves location update logic into the presenter (with tests)